org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/uriresolver/CacheResourceDownloadingException.java

```diff
@@ -20,46 +20,43 @@
  * Exception thrown when a resource (XML Schema, DTD) is downloading.
  *
  */
-public class CacheResourceDownloadingException extends RuntimeException {
+public class CacheResourceDownloadingException extends CacheResourceException {
 
 	private static final long serialVersionUID = 1L;
 
-	private static final String RESOURCE_LOADING_MSG = "The resource ''{0}'' is downloading.";
+	public enum CacheResourceDownloadingError {
 
-	private static final String RESOURCE_NOT_IN_DEPLOYED_PATH_MSG = "The resource ''{0}'' cannot be downloaded in the cache path.";
-	
-	private final String resourceURI;
+		DOWNLOAD_FORBIDDEN("Download is forbidden."), //

```

Comment:
I would consider renaming the message :

`Downloading external resources is disabled`

I'm hesitant to say `forbidden` because the language server only knows that `xml.downloadExternalResource` setting was sent as false. It has no knowledge of whether it was done for security reasons (eg. workspace trust) or whether the user simply changed the setting themselves.

Reply:
fixed